android-components/components/browser/engine-gecko/src/test/java/mozilla/components/browser/experiment/NimbusExperimentDelegateTest.kt

```diff
@@ -0,0 +1,71 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.browser.engine.gecko.activity
+
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import mozilla.components.experiment.NimbusExperimentDelegate
+import mozilla.components.support.test.mock
+import mozilla.components.support.test.whenever
+import org.hamcrest.CoreMatchers.equalTo
+import org.hamcrest.MatcherAssert.assertThat
+import org.json.JSONObject
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mozilla.geckoview.ExperimentDelegate
+import org.mozilla.geckoview.GeckoResult
+import org.mozilla.geckoview.GeckoRuntime
+import org.mozilla.geckoview.GeckoRuntimeSettings
+
+@RunWith(AndroidJUnit4::class)
+class NimbusExperimentDelegateTest {
+
+    @Test
+    fun `WHEN an experiment delegate is set on the runtime THEN the experiment delegate responds`() {
+        class MockExperimentDelegate : ExperimentDelegate {
+            override fun onGetExperimentFeature(feature: String): GeckoResult<JSONObject>? {
+                val result = GeckoResult<JSONObject>()
+                if (feature == "test") {
+                    result.complete(JSONObject().put("item-one", true).put("item-two", 5))
+                    return result
+                }
+                return null
+            }
+
+            override fun onRecordExposureEvent(feature: String): GeckoResult<Boolean>? {
+                val result = GeckoResult<Boolean>()
+                if (feature == "test") {
+                    result.complete(true)
+                } else {
+                    result.complete(false)
+                }
+                return result
+            }
+        }
+        val runtime = mock<GeckoRuntime>()
+        val mockExperimentDelegate = MockExperimentDelegate()
+        val mockGeckoSetting = mock<GeckoRuntimeSettings>()
+        whenever(runtime.settings).thenReturn(mockGeckoSetting)
+        whenever(mockGeckoSetting.experimentDelegate).thenReturn(mockExperimentDelegate)
+        val experimentDelegate = runtime.settings.experimentDelegate
+        experimentDelegate?.onGetExperimentFeature("test")?.accept {
+            assertThat("Experiment Item One Matches", it?.get("item-one"), equalTo(true))
+            assertThat("Experiment Item Two Matches", it?.get("item-two"), equalTo(5))
+        }
+        experimentDelegate?.onRecordExposureEvent("test")?.accept {
+            assertThat("Experiment Recorded an Exposure Event", it, equalTo(true))
+        }
```

Comment:
IIUC: this is just testing the `MockExperimentDelegate`?

Reply:
Yeah, now that you mention it, I kinda covered testing a generic `ExperimentDelegate` delegate in GeckoView. I updated this test to only check if everything was setting on the runtime as expected to simplify.